socorro/unittest/processor/rules/test_mozilla.py

```diff
@@ -1384,90 +1384,100 @@ def test_aurora_channel(self):
+    def test_lsb_release(self):
+        # If this is Linux and there's data in json_dump.lsb_release.description,
+        # use that
+        raw_crash = copy.deepcopy(canonical_standard_raw_crash)
+        raw_dumps = {}
+        processor_meta = get_basic_processor_meta()
 
-        rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "OS X 10.18"
+        rule = OSPrettyVersionRule()
 
-        # An invalid version of Mac OS X.
-        processed_crash = DotDict()
-        processed_crash.os_name = "Mac OS X"
-        processed_crash.os_version = "12.1"
+        processed_crash = {
+            "os_name": "Linux",
+            "os_version": "0.0.0 Linux etc",
+            "json_dump": {"lsb_release": {"description": "Ubuntu 18.04 LTS"}},
+        }
 
         rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "OS X Unknown"
+        assert processed_crash["os_pretty_version"] == "Ubuntu 18.04 LTS"
 
-        # Any version of Linux.
-        processed_crash = DotDict()
-        processed_crash.os_name = "Linux"
-        processed_crash.os_version = "0.0.12.13"
+    @pytest.mark.parametrize(
+        "os_name, os_version, expected",
+        [
+            ("Linux", None, "Linux"),
+            (None, None, None),
+            ("Windows NT", "NaN", "Windows NT"),
+        ],
+    )
+    def test_junk_data(self, os_name, os_version, expected):
+        raw_crash = copy.deepcopy(canonical_standard_raw_crash)
+        raw_dumps = {}
+        processor_meta = get_basic_processor_meta()
 
-        rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "Linux"
+        rule = OSPrettyVersionRule()
 
         # Now try some bogus processed_crashes.
-        processed_crash = DotDict()
-        processed_crash.os_name = "Lunix"
-        processed_crash.os_version = None
+        processed_crash = {}
+        if os_name is not None:
+            processed_crash["os_name"] = os_name
+        if os_version is not None:
+            processed_crash["os_version"] = os_version
 
         rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "Lunix"
+        assert processed_crash["os_pretty_version"] == expected
 
-        processed_crash = DotDict()
-        processed_crash.os_name = None
-        processed_crash.os_version = None
-
-        rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] is None
+    def test_dotdict(self):
```

Comment:
Since I rewrote it to use dicts, we need to make sure it works with `DotDict` still until we remove that.